dialogue-serde/src/main/java/com/palantir/conjure/java/dialogue/serde/ConjureBodySerDe.java

```diff
@@ -218,11 +218,10 @@ public T deserialize(Response response) {
                 if (errorDecoder.isError(response)) {
                     throw errorDecoder.decode(response);
                 } else if (response.code() == 204) {
-                    if (!isOptionalType) {
-                        throw new SafeRuntimeException(
-                                "Unable to deserialize non-optional response type from 204", SafeArg.of("type", token));
-                    }
-                    return TypeMarkers.getEmptyOptional(token);
+                    // TODO(dfox): what if we get a 204 for a non-optional type???
+                    // TODO(dfox): support http200 & body=null
```

Comment:
These last two cases are handled by the regular deserializer

Reply:
In most cases that's correct, unfortunately null->empty collection won't work that way because we can't provide field annotations.